### ISSUE TITLE
Regression: become prompt not recognized, leading to timeout errors

### ISSUE THREAD

**Issue by sbaudoin** [ansible/ansible#24462]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
connection plugin

##### ANSIBLE VERSION
```
ansible 2.3.0.0
  config file = /root/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.5 (default, Aug  2 2016, 04:20:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-4)]
```

##### CONFIGURATION
`ansible.cfg`:
```
[defaults]
host_key_checking = False
#timeout = 30

[ssh_connection]
ssh_args = -o ControlMaster=no
```

##### OS / ENVIRONMENT
RHEL 7.2, Python 2.7.5

##### SUMMARY
With Ansible 2.3.0.0 it is no longer possible to use ansible_become_pass and ansible_become_method=su. We systematically get the following error:
```
Timeout (12s) waiting for privilege escalation prompt:
```

After some investigations it appears that the prompt function `detect_su_prompt` defined in `playbook/play_context.py` around the line 530 is no longer called via the method `check_password_prompt` in `plugins/connection/__init__.py`. If we compare with the code of the version 2.2 there was an else statement that called the prompt function that no longer exists in 2.3.0.0. If we introduce again the else statement, the escalation prompt is successfully recognized again.

##### STEPS TO REPRODUCE
Inventory file `host.ini`:
```yaml
[all:vars]
ansible_connection=ssh
ansible_user="my_login"
ansible_ssh_pass="my_password"
ansible_become=true
ansible_become_method=su
ansible_become_pass="my_root_password"

[targets]
my_server
```

Then I call:
```
ansible my_server -m shell -a "/sbin/vgdisplay" -i host.ini -b
```

##### EXPECTED RESULTS
With the example above I expect to get the list of volume groups of my machine.

##### ACTUAL RESULTS
I systematically get a timeout error:
```
Timeout (12s) waiting for privilege escalation prompt:
```

**Comment by abadger** [comment]
This should be fixed by: https://github.com/ansible/ansible/pull/24081  which is in the current devel and stable-2.3 branches (for release with 2.3.1)

Are you able to confirm that the current release candidate: http://releases.ansible.com/ansible/ansible-2.3.1.0-0.1.rc1.tar.gz or a current checkout of the stable-2.3 branch fixes the problem?

needs_info

**Comment by sbaudoin** [comment]
Hello,

Yes, thanks, this works with this 2.3.1.0-0.1 RC1.

**Comment by abadger** [comment]
hanks for testing!  Closing.  Will be fixed in 2.3.1.
